### PR TITLE
Statsreset v2

### PR DESCRIPTION
**Pull request by treymd**

### CODE REVIEW

**Comment by skittles1** [comment]
A recent change seems to have broken stat reset, to save you the trouble of tracking it down here are the error messages from 104 (only second one is relevant I guess):

Feb 18 2015 09:34:45|[user.bof (1482)] OBJECT 21615,sent unknown command from client,157

Feb 18 2015 09:34:45|InterpretAtMessage found too many locals and parms for OBJECT 21615 CLASS User MESSAGE UserChangedStats (user.bof (9146)) aborting and returning NIL 

Wasn't aware there was a limit that included locals (related to the 15 parameter limit perhaps?). Might have to split some or most of that message up.

**Comment by treymd** [comment]
Yes, there appears to be both a 15 item limit on parameters, as well as a 15 item limit on locals.  I removed the parameter I just added to hold the penalty queried from settings.kod.  I am now at 14 params, and 15 locals.  I really would like to avoid any major changes to this message right now, but it is at its limits.

File: kod/object/passive/skill.kod [hunk 1]
```diff
@@ -181,6 +181,48 @@ messages:
...
+         OR (viSkill_Num = SKID_PROFICIENCY_AXE)
+      {
+         return STAT_ID_AGILITY;
+      }
+
+      if (viSkill_Num = SKID_FIRE)
+         OR (viSkill_Num = SKID_PROFICIENCY_SWORD)
+         OR (viSkill_Num = SKID_PROFICIENCY_BOW)
+         OR (viSkill_Num = SKID_PROFICIENCY_SHORT_SWORD)
+         OR (viSkill_Num = SKID_PROFICIENCY_SHORTSWORD)
```
Comment (skittles1) [inline kod/object/passive/skill.kod]: Only need one of the shortsword proficiencies here (same ID number)

File: kod/object/passive/spell/crysmana.kod [hunk 2]
```diff
@@ -196,5 +196,10 @@ messages:
       return TRUE;
    }
 
+   CanForget()
```
Comment (skittles1) [inline kod/object/passive/spell/crysmana.kod]: Don't need a CanForget override here, the default value in spell.kod is TRUE

**Pull request merged**